### DISCUSSION TITLE
Integration questions

### DISCUSSION THREAD

**Discussion by tbillington** [Byron/trash-rs#77] [Q&A] [answered]
Hey! I'm a maintainer of [kondo](https://github.com/tbillington/kondo), a tool to delete artifacts like `target` and `node_modules`.

I'm [looking into](https://github.com/tbillington/kondo/issues/96) supporting trashing as an option. Kondo's current behaviour is a recursive delete.

I was wondering if there would be any particular behaviours or issues you foresee when integrating `trash-rs`. For example this [open issue around deleting non-empty directories on windows](https://github.com/Byron/trash-rs/issues/70) would likely be a blocker. I'm happy to help investigate where I can for similar issues.

I'm also curious about the performance difference between deletion of a large directory with many files and moving it to trash, do you have any insight here?

Hope this location is okay to ask these questions :) otherwise I'm happy to talk in https://github.com/tbillington/kondo/issues/96 too.

**Comment by Byron** [comment, marked as answer]
Thanks for considering using `trash-rs` :)!

The issue you find actually has been resolved already (and just recently), and was just left-over. With that said, on windows it can't delete a single folder right now, but will instead delete each file recursively. This is certainly not optimal for performance, and maybe has another, better solution. The current one is clearly just at the level of 'getting it to work'.

> I'm also curious about the performance difference between deletion of a large directory with many files and moving it to trash, do you have any insight here?

On unix, any directory is just a single item, but on windows it's the entire contents. I also don't know if the test-suite covers the restoration of all of these recursively deleted files, or how this looks in the windows UI. I recommend trying the examples to see for yourself.

**Reply by tbillington** [reply]
The recursive delete is probably fine, at the end of the day that's what [kondo does at the moment](https://github.com/tbillington/kondo/blob/a7af95484d364bbb12eb3b40b0d860424dd1b714/kondo-lib/src/lib.rs#L236), but I push it to a background thread. Also SSDs are fast :) 

Moving a directory to trash on linux is probably actually way faster than the recursive delete if it is just a filesystem move, which would actually be a big performance win.
